### ISSUE TITLE
prefer-template: Add config to restrict series of string concatinations

### ISSUE THREAD

**Issue by apexskier** [palantir/tslint#3034]
It looks like https://github.com/palantir/tslint/pull/2741 allows a series of string concatenations, which I'd like to disable in our codebase. The specific use case was someone using the `+` operator to concatenate a string with a template literal, when it would be preferred to use a single template literal. Since it looks like this was an intentional change, I'd like to add configuration to `prefer-template` to work around this.

simple proposal
```
{
  "type": "array",
  "enum": [
    "allow-single-concat",
    "allow-pure-string-concat" // enables the change introduced in #2741
  ]
}
```

### Bug Report

- __TSLint version__: 5.4.3 (introduced in 5.3.0)
- __TypeScript version__: 2.3.4
- __Running TSLint via__: any

#### TypeScript code being linted

```ts
const world = "world";

console.log("hello" + " world"); // a
console.log("hello" + ` ${world}`); // b
console.log("hello " + world); // c
```

with `tslint.json` configuration:

```json
{
    "rules": {
        "prefer-template": true
    }
}
```

#### Actual behavior
a: no warning
b: no warning
c: warning

#### Expected behavior
a: no warning
b: warning
c: warning

**Comment by JoshuaKGoldberg** [comment]
☠️ *TSLint's time has come!* ☠️ 

TSLint is no longer accepting most feature requests per #4534. See **[typescript-eslint.io](https://typescript-eslint.io)** for the new, shiny way to lint your TypeScript code with **ESLint**. ✨ 

It was a pleasure open sourcing with you all!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
